### PR TITLE
Draft: Payment for Projects

### PR DESCRIPTION
**Pull request by jacoblb64**
Looking for some feedback and some help testing with this!

Work Completed
- Handling both projects and sellers for the /charges endpoint
- Dealing with all the logic for webhooks
- Misc. Migrations annotations that were automatically updated

Testing Done
- Testing simple happy path for a payment with a project ID and no seller ID

Outstanding Work
- Update unit tests
- Test all existing functionality

### CODE REVIEW

File: app/controllers/webhooks_controller.rb [hunk 1]
```diff
@@ -193,14 +212,16 @@ def handle_payment_intent_succeeded(
...
       grouped_items.each_key do |sid|
-        EmailManager::DonationReceiptSender.call(
-          {
-            payment_intent: payment_intent,
-            amount: grouped_items[sid].map { |li| li['amount'].to_f }.sum,
-            merchant: Seller.find_by(seller_id: sid).name,
-            email: payment_intent.purchaser.email
-          }
-        )
+        if sid.present?
```
Comment (justintmckibben) [inline app/controllers/webhooks_controller.rb:215]: This means that we don't send emails for light up chinatown donatoins?

Comment (jacoblb64) [inline app/controllers/webhooks_controller.rb:215]: as seller_id can now be null

File: app/services/webhook_manager/item_creator.rb [hunk 2]
```diff
@@ -3,23 +3,35 @@
...
+          campaign_id: payment_intent.campaign_id
+        )
+      else
+        Item.create!(
+          project: project,
+          purchaser: payment_intent.purchaser,
+          item_type: item_type,
+          payment_intent: payment_intent
+        )
+      end
```
Comment (justintmckibben) [inline app/services/webhook_manager/item_creator.rb:34]: I wonder if we should raise if neither seller nor project exist

File: app/controllers/webhooks_controller.rb [hunk 3]
```diff
@@ -141,6 +142,24 @@ def handle_payment_intent_succeeded(
...
+        WebhookManager::DonationCreator.call(
+          {
+            seller_id: seller_id,
+            project_id: project_id,
+            payment_intent: payment_intent,
+            amount: amount
+          }
+        )
+
+        EmailManager::DonationReceiptSender.call(
```
Comment (jacoblb64) [inline app/controllers/webhooks_controller.rb:155]: not setting `is_donation` and instead sending a one-off email from here--could be customized further too

**Review by justintmckibben — APPROVED** [review]
Looks good! Nothing blocking

File: app/services/webhook_manager/item_creator.rb (hunk 2 above)
Comment (jacoblb64) [reply app/services/webhook_manager/item_creator.rb:34]: Was gonna mention the same thing, for this and some other places, if we should have a default case where neither is there and throw an exception

File: app/controllers/webhooks_controller.rb (hunk 1 above)
Comment (jacoblb64) [reply app/controllers/webhooks_controller.rb]: `seller_id` can be null now, when there is a `project_id`, so we should skip it here (the grouping logic seemed weird, and it should be on it's own anyway). I'm triggering a separate email in the case above for a project

File: app/services/webhook_manager/item_creator.rb (hunk 2 above)
Comment (Jeremy-Xue) [reply app/services/webhook_manager/item_creator.rb:34]: tiny bit of overlap with what I was working on here, but this is basically what I had!

**Pull request merged**